src/armature/Import.ts

```diff
@@ -26,30 +26,51 @@ class Data {
         this.lines = rawData.split('\n');
     }
 
-    public readLine() {
-        let line = this.lines[0].split(' ');
-        this.lines.shift();
-        // Skip comments and empty lines.
-        while (line.length < 1 || line[0].startsWith('#')) {
-            line = this.lines[0].split(' ');
-            this.lines.shift();
-        }
-
-        return line;
-    }
-
-    // TODO: Refactor to handle each line type individually and multipblex
-    // (so we're not reliant on a particular order)
-    public getLinesWithPrefix(prefix: string) {
-        const lines: string[][] = [];
-        while (this.lines[0].startsWith(prefix)) {
-            lines.push(this.readLine());
-        }
-
-        return lines;
+    public read() {
+        const vertices: vec3[] = [];
+        const normals: vec3[] = [];
+        const groups: Group[] = [];
+        let materialName = '';
+
+        while (this.lines.length > 0) {
+            const [command, ...args] = (<string>this.lines.shift()).split(/\s+/);
+
+            if (command === 'v') {
+                const vecLine = args.map(parseFloat)
+                const vec = vec3.fromValues(vecLine[0], vecLine[1], vecLine[2]);
+                vertices.push(vec);
+            } else if (command === 'vn') {
+                const vecLine = args.map(parseFloat)
+                const vec = vec3.fromValues(vecLine[0], vecLine[1], vecLine[2]);
+                normals.push(vec);
+            } else if (command === 'g') {
+                groups.push({ groupName: args[0], materialName, faces: [] });
```

Comment:
Is it okay if `materialName` is empty string at this point or should we throw an exception?

Reply:
It's ok if it's an empty string here because we could do a group and only then specify the material on the next line. It'd be bad if we never set the material though, so I can add an assertion later when we're getting a concrete `Material` for each group.